### ISSUE TITLE
[Bug] description

### ISSUE THREAD

**Issue by bphd** [chaotic-aur/packages#2726]
### Packages:

Something's not clear in your homepage. You tell that you have 

### Latest chaotic version

banned packages, you tell that you are the pre-built version of AUR, you tell that it's updated hourly  

### What happens?

So from reading that, I come to the conclusion that Chaotic have ALL AUR packages, and if not, it will the next hour (time of update). But it seems in fact to not have them all, and when you come to GitHub

### What is expected to happen?

, there is even a request section. So packages are not all available, it's not AUR pre-built. It's just a selection of packages that you build, yeah you take AUR PKGBUILD to have instructions to build it, but it's not AUR pre-built

### More information

So two requests: 1) Why it's not full-AUR? Why not making it? Which criterion to select a package or not?; 2) Explain better that part in the homepage

**Comment by FabioLolix** [comment]
> but it's not AUR pre-built

In the Chaotic repo there are the prebuilt packages

Commiting the builded packages here does not make sense if it what you mean

> Which criterion to select a package or not?

https://aur.chaotic.cx/

`This repo is generated on-demand from packages we use on our personal computers. If there are any package license infringements please notify us through email. The package will then be removed and blacklisted.` 

> Why it's not full-AUR?

On AUR as of today there are 85677 'Packages' (sic)

On all Arch repos there are 14359 Packages

Building and supervisioning ~85000 pkgbuilds is a huge task in every aspect: time, computational power, storage, etc... especially in your free time and for free

**Comment by bphd** [comment]
> Commiting the builded packages here does not make sense if it what you mean

I don't understand what this sentence mean, the concern I want to talk about, you talk about it in your 3rd reply

> https://aur.chaotic.cx/

I thought you said that to justify the personal use of any package that has license infringements, to not be incriminated. Didn't understand that it was meant to describe the criterion of what is on chaotic. If this is the criterion then you use way too many packages as chaotic is tremendously big for one computer/user, even a group of maintainer. If the `real` criterion is more about saying yes/no to the requests made over here, what's the use? It's not a specific criterion, it's only a time bottleneck

> Building and supervisioning ~85000 pkgbuilds is a huge task in every aspect: time, computational power, storage, etc... especially in your free time and for free

Sure thing, I understand. So why privileging some packages over others? Why not reducing refreshing time over package support? If you multiply by two the compiling time, you'll have on average resource for two times already maintained package, needing no more resources to be available. I think community will prefer way much 28718 packages every two hours, or even 85677 packages every six hours, than aggressive refresh time. If you use cloud, so you pay per cycle used rather than time allocated to a resource, then it's another problem. But even here, it is still better to compile different packages than the same package six times, use the same resources

**Comment by FabioLolix** [comment]
> 
> Something's not clear in your homepage. You tell that you have banned packages, you tell that you are the pre-built version of AUR, you tell that it's updated hourly 
> 
> So from reading that, I come to the conclusion that Chaotic have ALL AUR packages, and if not, it will the next hour (time of update). But it seems in fact to not have them all, and when you come to GitHub, there is even a request section. So packages are not all available, it's not AUR pre-built.

Let's take a step back, the website indicate that:

`Automated building repo for [AUR](https://aur.archlinux.org/) packages`

About

> Feel free to create issues to suggest enhancements, flag database corruptions and all other requests: [Issues](https://github.com/chaotic-aur/packages/issues)

`all other requests` imply request for new packages

and the disclaimer at the bottom:

`This repo is generated on-demand from packages we use on our personal computers.`

So it don't have all the AUR packagaes

> So packages are not all available, it's not AUR pre-built.

No not all packages are available, `it's not AUR pre-built` not sure what you mean with that

***

> Sure thing, I understand. So why privileging some packages over others?

since the number of pkgbuild is massive start small and then ass what the community ask if it is doable

note that downloading/checking all the pkgbuilds over a short period of time is not possible, the server will block the access

**Comment by xiota** [comment]
@bphd 

> 1) Why it's not full-AUR? Why not making it?

There are not enough resources for Chaotic to build the entire AUR.  Even if there were, many packages are broken, conflicting with each other, or have other significant issues.

> Which criterion to select a package or not?

Depends on each maintainer.  Some may add all requests.  Others may add only packages that they are interested in.

> 2) Explain better that part in the homepage

@dr460nf1r3 Perhaps update chaotic.cx to say "Automated building repo for **selected** AUR packages"

**Comment by bphd** [comment]
> > Something's not clear in your homepage. You tell that you have banned packages, you tell that you are the pre-built version of AUR, you tell that it's updated hourly
> > So from reading that, I come to the conclusion that Chaotic have ALL AUR packages, and if not, it will the next hour (time of update). But it seems in fact to not have them all, and when you come to GitHub, there is even a request section. So packages are not all available, it's not AUR pre-built.
> 
> Let's take a step back, the website indicate that:
> 
> `Automated building repo for [AUR](https://aur.archlinux.org/) packages`
> 
> About
> 
> > Feel free to create issues to suggest enhancements, flag database corruptions and all other requests: [Issues](https://github.com/chaotic-aur/packages/issues)
> 
> `all other requests` imply request for new packages
> 
> and the disclaimer at the bottom:
> 
> `This repo is generated on-demand from packages we use on our personal computers.`
> 
> So it don't have all the AUR packagaes
> 
> > So packages are not all available, it's not AUR pre-built.
> 
> No not all packages are available, `it's not AUR pre-built` not sure what you mean with that

In your first part you try to prove that it's possible to interprate based on the homepage that not all packages are available. I don't ask you if it's possible to interprete that, I ask you to make it clearer on the homepage, to be more precise, to precise it explicitely



> since the number of pkgbuild is massive start small and then ass what the community ask if it is doable
> 
> note that downloading/checking all the pkgbuilds over a short period of time is not possible, the server will block the access

Why doing that instead of supporting them all best effort or just by popularity best effort?

Why the server will not block for 10k+ but not 60k+? Like I've told earlier it's really much more useful to have all packages, even one week old, than not having them. I will not die having a package from 12 august. But it's a pain not having at all a package on chaotic when this package take tremendous ressources to make

If it's about downloading pkgbuild, keep respecting the limit but download them all, it will be 6 times slower, but I mean, you're already competitive with one hour refresh, it's okay

**Comment by bphd** [comment]
> not enough resources

If you can build 10k packages per hour, then you can build 60k per 6 hours. It requires the same resources per hour. You actually do build 60k per 6 hour... It's just 6 times the same packages... (maybe you optimize by building only new pkdbuild)



> packages are broken

Yeah, aswell as many that you support already. Packages break everyday, sometimes being repaired, sometimes not until the end of the time. Chaotic still run with that. And if a package is broken in aur/ it will be on /chaotic too, you're not obliged to by more reliable than /aur. It's okay if no package are published because broken, or even that broken one is published. It will give the same result as trying to build from /aur. It will just eat bandwidth instead of CPU

> Depends

Okay so it's like on AUR, people manage packages. I thought chaotic was only a server executing dumbly all pkgbuild so it doesnt need human resources that much. I would personally prefer a fully automated very broken with all packages chaotic, than a tiny curated list, because I need many packages. If it's broken, it's up to AUR maintainer to do a pkgbuild that works, not to chaotic to make anything but compile

> selected

Maybe a more precise term, less vague. Like "curated list", "maintainer selection", idk, something that explain clearly that chaotic doesnt host whatever we want, only what maintainer personally choose upon personal criteria. It's important I think

**Comment by bphd** [comment]
> @[xiota](https://github.com/xiota) xiota added the[ reject:too-hard-for-us ](https://github.com/chaotic-aur/packages/labels/reject%3Atoo-hard-for-us)label [Aug 19, 2023](https://github.com/chaotic-aur/packages/issues/2726#event-10136736506)

You labeled as rejected but you actually try to resolve the problem, trying to update the homepage. Don't be shy to make better yearly issues statistics by labeling as completed/done/successful when done

**Comment by Technetium1** [comment]
@bphd We made it more than a couple of thousand issues before there was confusion. I don't believe it's urgent to update the phrasing of the website. If one group could build everything, then there wouldn't be [so much choice](https://wiki.archlinux.org/title/unofficial_user_repositories).

> I would personally prefer a fully automated very broken with all packages chaotic, than a tiny curated list, because I need many packages. If it's broken, it's up to AUR maintainer to do a pkgbuild that works, not to chaotic to make anything but compile

I don't think that this is not the solution for you. I recommend you use https://github.com/alexheretic/aurto or similar instead. You can always have [your own repository](https://wiki.archlinux.org/title/Pacman/Tips_and_tricks#Custom_local_repository) with latest builds from AUR. There are also [plenty of other options](https://wiki.archlinux.org/title/AUR_helpers).

Otherwise, please open _individual_ issues and follow the template as requested in https://github.com/chaotic-aur/packages/issues/2727#issuecomment-1685075694.

**Comment by bphd** [comment]
> I don't believe it's urgent 

I don't said it has to be done in the day. Just that its a good thing to do when possible. You can't just close things that are not urgent, there is a priority ticketing system, why not just set low priority and let open?

**Comment by bphd** [comment]
> there wouldn't be so much choice.

It's not the reason why there is multiple repo. There is multiple because they want their own rules on it, not to share the burden of compiling part of AUR. It'd more chaotic wish to be a repo that compile AUR PKGBUILD

**Comment by bphd** [comment]
> I recommend you

If I had the ability to decently compile I wouldn't have the need to search for a pre compiled repo. You advise to someone who has no water to get water. I search for a place to download binaries, not different ways of compiling packages

**Comment by bphd** [comment]
> requested in 

You're supposed to adress concerns about an issue... on that issue page, not advising randomly on page where the concerned person is hanging. And you repeat something that has already been said on the dedicated page, and there was even an answer to it. What's the interest into doing that? It gives no more information whatsoever

**Comment by dr460nf1r3** [comment]
> Perhaps update chaotic.cx to say "Automated building repo for _selected_ AUR packages" (or similar)

> I don't ask you if it's possible to interprete that, I ask you to make it clearer on the homepage, to be more precise, to precise it explicitely

Is this not what you were trying to achieve here?

I'm quite uncertain about what the entire discussion here is about. 
- Building all AUR packages vs selecting packages to build? I think it's quite obvious that building the entire AUR does _NOT_ work as too many packages are outdated, broken, or unmaintained. Therefore, a selection has to be made either way.
- Compiling more packages less often vs compiling fewer packages more often - in theory, I agree, but we are rather limited by the mirror's disk space than actual missing CPU power. 
- Updating the phrasing of the homepage to be more accurate?

**Comment by bphd** [comment]
> Is this not what you were trying to achieve here?

What I try to achieve is to make it clearer on the homepage. It's not meant to be achieved here but on the homepage

> a selection has to be made

It's not about building it all, it's about trying to. If it fails, no problem, we have the last successful build and we'll wait till the pkgbuild is repaired. Many package switch from "out to date" to active everyday in both way, it's pretty normal that not everything builds. But if we don't try, the number of 10k will never go significantly up

> mirror's disk space

If there is multiple mirrors, then reducing the redundency and sharing the load of all the packages, so more can be hosted. Always the same problem about how resources are used, how many times for the same element

> Updating the phrasing of the homepage to be more accurate?

Yup yup, made some suggestions up here that I feel to be clear, don't hesitate to re formulate

And little suggestion, don't label "won't do it" and close as not planned request that you are actually treating, (neither closing as completed when not). I get that sometimes people don't like request or how they're made. But such tools are to describe the process status, not to transmit a feeling about the situation

**Comment by bphd** [comment]
> There is also a package list on the website if unsure about the content of Chaotic-AUR:

Yeah, another UI issue, when I opened the website I used the search bar but it was website wide, not chaotic-wide, lol. But hey, it's their problem and Ctrl+F do the work
